instrumentedProfilingLibrary/Time Profiler/ProfileDataManager.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <string>
+#include <chrono>
+#include <fstream>
+
+class ProfileDataManager {
+	
+public:
+    static ProfileDataManager* GetInstance(const std::string& name);
+    ProfileDataManager(ProfileDataManager& other) = delete;
+    void operator=(const ProfileDataManager&) = delete;
+    
+    void displayTimerData();
+    void storeTimerData();
+    void getTimerData(const std::string& name, std::chrono::microseconds timetaken, std::chrono::time_point<std::chrono::high_resolution_clock> codeBlockStartTime,
+        std::chrono::time_point<std::chrono::high_resolution_clock> codeBlockEndTime);
+
+private:
+    std::string codeBlockname;
+    uint64_t StartTimeStamp_us;

```

Comment:
If you measure more than one code block, then you kinda lose the previous measurements.

Reply:
should I use arrays here to store consecutive measurements? I had not figured this out and since I am able to display the measurements on the screen, I had  thought that nothing is lost.  Another thing that I have observed is that the measurements vary with big margins from one execution to the next  and its not possible to say 'code block x' takes  about 'n' microseconds.